src/main/resources/META-INF/plugin.xml

```diff
@@ -430,7 +431,34 @@
                 implementation="com.redhat.devtools.lsp4ij.features.semanticTokens.LSPSemanticTokensHighlightVisitor"/>
         <colorSettingsPage
                 implementation="com.redhat.devtools.lsp4ij.features.semanticTokens.SemanticTokensColorSettingsPage"/>
+    </extensions>
 
+    <!-- Semantic tokens-based file view provider -->
+    <extensions defaultExtensionNs="com.intellij">
+        <!-- File view provider based on textDocument/semanticTokens -->
+        <!--
+         NOTE: This can only be registered for TextMate files because abstract file type plain text files have no
+         language and a variable file type.
+         -->
+        <lang.fileViewProviderFactory
+                language="textmate"
+                implementationClass="com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokensFileViewProviderFactory"/>
+        <lang.elementManipulator
+                forClass="com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenPsiElement"
+                implementationClass="com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenPsiElementManipulator"/>
+        <lang.documentationProvider
+                language=""
+                implementationClass="com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenDocumentationProvider"/>
+    </extensions>
+
+    <!-- Register the semantic tokens file view provider for the CSS/JS abstract file types in Community Edition -->
+    <!--suppress PluginXmlValidity -->
+    <depends optional="true" config-file="plugin-file-view-provider-ce.xml">com.intellij.modules.idea.community</depends>
+    <!--suppress PluginXmlValidity -->
+    <!-- TODO: Disabling until https://platform.jetbrains.com/t/686 is resolved -->
+    <!--<depends optional="true" config-file="plugin-file-view-provider-ce.xml">com.intellij.modules.pycharm.community</depends>-->
```

Comment:
Note that the PyCharm CE optional dependency is disabled until I get an answer to [this question](https://platform.jetbrains.com/t/686). I have this exact optional dependency in my own plugin in a few places and pass the plugin verifier fine, but my plugin uses an older Ant-based build instead of Gradle. Hopefully I'll get a quick and simple answer.